docs/using-the-compiler.rst

```diff
@@ -0,0 +1,271 @@
+        //   evm.asm - Current assembly format (--asm)
+        //   evm.asmJSON - Current assembly format in JSON (--asm-json)
+        //   evm.opcodes - Opcodes list
+        //   evm.methodIdentifiers - The list of function hashes
+        //   evm.gasEstimates - Function gas estimates
+        //   evm.bytecode - Bytecode (--bin)
+        //   evm.deployedBytecode - Deployed bytecode (--bin-runtime)
+        //   evm.sourceMap - Source mapping (useful for debugging)
+        //   ewasm.wast - eWASM S-expressions format (not supported atm)
+        //   ewasm.wasm - eWASM binary format (not supported atm)
+        outputSelection: {
+          // Enable the metadata and bytecode outputs of every singel contract.
+          "*": {
+            "*": [ "metadata", "evm.bytecode" ]
+          },
+          // Enable the abi and opcodes output of MyContract defined in file def.
+          "def": {
+            "MyContract": [ "abi", "evm.opcodes" ]
+          },
+          // Enable the source map output of every single contract.
+          "*": {
+            "*": [ "evm.sourceMap" ]
+          },
+          // Enable the AST and Why output of every single file.
+          "*": {
+            "": [ "ast", "why3" ]
+          }
+        }
+      }
+    }
+
+
+Output Description
+------------------
+
+.. code-block:: none
+
+    {
+      // Optional: not present if no errors/warnings were encountered
+      errors: [
+        {
+          // Optional
+          file: "sourceFile.sol",
+          // Optional: Location within the source file.
+          sourceLocation: [
+            { start: 0, end: 100 },
+          ],
+          // Mandatory: Error type, such as "TypeError", "InternalCompilerError", "Exception", etc
+          type: "TypeError",
+          // Mandatory: Component where the error originated.
+          component: "general", "why3", "ewasm", etc.
+          // Mandatory
+          severity: "warning" or "error",
+          // Mandatory
+          message: "Invalid keyword"
+        }
+      ],
+      // This contains the file-level outputs. In can be limited/filtered by the outputSelection settings.
+      sources: {
+        "sourceFile.sol": {
+          // Identifier (used in source maps)
+          id: 1,
+          // The AST object
+          ast: {}
+        }
+      },
+      // This contains the contract-level outputs. It can be limited/filtered by the outputSelection settings.
+      contracts: {
+        "sourceFile.sol": {
+          "ContractName": {
+            // The Ethereum Contract ABI. If empty, it is represented as an empty array.
+            // See https://github.com/ethereum/wiki/wiki/Ethereum-Contract-ABI
+            abi: [],
+            evm: {
+              // Intermediate representation (string)
+              ir: "",
+              // Assembly (string)
+              assembly: "",
+              // Old-style assembly (string)
```

Comment:
I don't think we will have this anymore. The layout is already almost new-style assembly.